### ISSUE TITLE
XCrypto Repository management todo list.

### ISSUE THREAD

**Issue by ben-marshall** [scarv/xcrypto#73]
**Context:**

When the project first started, *this* repository was a mono-repo containing the specification, reference implementation and a good deal of software. Before the `v0.15.0` release, we split the repository such that it acted as a "super-repo", with submodule pointers to the relevent components. This made sense at the time in order to allow development of the specification independently of the reference implementation while we were changing things rapidly. This was doubly true when we developed the reference implementation as a co-processor, rather than a now where it is tightly integrated into a CPU.

**Changes needed:**

Now we are at version `v1.0.0`, it makes sense to update the repository structure again:
- [x] Merge the specification from scarv/xcrypto-spec back into this repository.
- [x] Merge the example RTL from scarv/xcrypto-rtl into this repository.
- [x] Archive / mothball the old scarv/xcrypto-ref implementation, since it implements the old `v0.15.0` branch specification. Delete the submodule reference from this repository.
- [x] Add submodule links to the scarv/scarv-cpu and scarv/scarv-soc repositories, which implement the `v1.0.0` branch of XCrypto.
- [x] Adopt a patch-based method of adding `bintutils` and `spike` changes, similar to how riscv/riscv-bitmanip works.
  - This will make it easier to keep repositories consistent wrt. submodule references. It will rely on picking a known-good binutils/spike version, and then some periodic work to merge upstream changes.
  - This will be annoying but not too painful, as the XCrypto changes are orthogonal to other work.

The end result of this work should see the scarv/xcrypto repository look and work much more like the riscv/riscv-bitmanip repository, and be much eaiser to manage as a result.

Future work on XCrypto, particularly with respect to the next iteration, will then be done on a dedicated `v2.0.0` branch and then merged back onto master.

All of the work in this issue *will* be done before we start on specification, software and hardware changes for the next iteration.

**Proposed Structure:**

```
xcrypto/            - Repository root.
- bin/              - Environment setup and tool scripts.
- doc/              - Project documentation
  - tex/            - LaTeX source for the specification document
  - presentations/  - Posters / slideshows etc.
- rtl/              - Reference implementations of the instructions.
- tools/            - Patches for the GCC toolchain and Spike simulator.
- docker/           - Docker container artifacts for pre-built toolchain etc.
- extern/           - Submodule references to related projects.
  - scarv-cpu       - Submodule reference to SCARV CPU project implementing XCrypto
  - scarv-soc       - Ditto for the SCARV SoC project which wraps the CPU.
  - libscarv        - Submodule reference to libscarv with example algorithms.
```

**Comment by danpage** [comment]
As discussed, agreed re. basically everything.  However, to align with other repos. I'd suggest tweaking the structure to something more like

```sh
xcrypto
├── bin                     - scripts (e.g., environment configuration)
├── build                   - working directory for build
├── doc                     - documentation, i.e., specification
├── extern                  - external resources (e.g., submodules)
│   ├── scarv-cpu             - submodule: scarv/scarv-cpu
│   ├──	scarv-soc             - submodule: scarv/scarv-soc
│   ├──	libscarv              - submodule: scarv/libscarv
│   └── texmf                 - submodule: scarv/texmf
├── media                   - presentations, slides, etc.
├── tools                   - patches for the tool-chain (e.g., GCC and Spike)
└── src
    ├── rtl                   - source code for
    ├── docker                - source code for containerised tool-chain
    └── example               - "hello world" example
```

**Comment by ben-marshall** [comment]
Closing as done a/o c48deeb
